### PR TITLE
Consumer group - fix race condition on creating state file

### PR DESCRIPTION
**Pull request by liranbg**
Upon state file creation, multiple replicas will try to create the file. we want to ensure only one succeed

### CODE REVIEW

**Review by omesser — APPROVED** [review]

File: pkg/dataplane/streamconsumergroup/streamconsumergroup.go [hunk 1]
```diff
@@ -145,16 +149,24 @@ func (scg *streamConsumerGroup) setStateInPersistency(state *State, mtime *int)
...
+
+		// mtime does not exist => file does not exist => create it
+		// we want the file to be created by one replica only and thus
+		// we condition the creation of it by checking if the state attribute
+		// does not exist
+		condition = fmt.Sprintf("not(exists(%s))", stateContentsAttributeKey)
 	}
 
-	_, err = scg.container.UpdateItemSync(&v3io.UpdateItemInput{
+	condition = fmt.Sprintf("not(exists(%s))", stateContentsAttributeKey)
```
Comment (assaf758) [inline pkg/dataplane/streamconsumergroup/streamconsumergroup.go:161]: why is this?

File: pkg/dataplane/streamconsumergroup/streamconsumergroup.go [hunk 2]
```diff
@@ -117,12 +117,16 @@ func (scg *streamConsumerGroup) setState(modifier stateModifier) (*State, error)
...
+				"mtime", mtime,
 				"previousState", previousState,
 				"modifiedState", modifiedState)
 		}
 
-		err = scg.setStateInPersistency(modifiedState, mtime)
-		if err != nil {
+		if err := scg.setStateInPersistency(modifiedState, mtime); err != nil {
+			if attempt%10 == 0 {
+				scg.logger.DebugWith("Failed to set state in persistency",
```
Comment (assaf758) [inline pkg/dataplane/streamconsumergroup/streamconsumergroup.go:127]: add `attempt` value

File: pkg/dataplane/streamconsumergroup/streamconsumergroup.go (hunk 1 above)
Comment (liranbg) [reply pkg/dataplane/streamconsumergroup/streamconsumergroup.go]: mistake

File: pkg/dataplane/streamconsumergroup/streamconsumergroup.go (hunk 2 above)
Comment (liranbg) [reply pkg/dataplane/streamconsumergroup/streamconsumergroup.go:127]: done

**Review by assaf758 — APPROVED** [review]

**Pull request merged**